apps/arweave/src/ar_nonce_limiter.erl

```diff
@@ -357,23 +356,30 @@ init([]) ->
 
 resolve_remote_server_raw_peers() ->
 	{ok, Config} = application:get_env(arweave, config),
-	Peers = resolve_remote_server_raw_peers(Config#config.nonce_limiter_server_trusted_peers),
-	ets:insert(?MODULE, {remote_servers, Peers}),
+	resolve_remote_server_raw_peers(Config#config.nonce_limiter_server_trusted_peers),
 	timer:apply_after(10000, ?MODULE, resolve_remote_server_raw_peers, []).
 
-resolve_remote_server_raw_peers(RawPeers) ->
-	resolve_remote_server_raw_peers(RawPeers, []).
-
-resolve_remote_server_raw_peers([], Peers) ->
-	Peers;
-resolve_remote_server_raw_peers([RawPeer | RawPeers], Peers) ->
+resolve_remote_server_raw_peers([]) ->
+	ok;
+resolve_remote_server_raw_peers([RawPeer | RawPeers]) ->
 	case ar_util:safe_parse_peer(RawPeer) of
 		{ok, Peer} ->
-			resolve_remote_server_raw_peers(RawPeers, [Peer | Peers]);
+			case ets:lookup(ar_nonce_limiter, {vdf_server_raw_peer, RawPeer}) of
+				[] ->
+					ets:insert(ar_nonce_limiter, {{vdf_server_raw_peer, RawPeer}, Peer}),
+					ets:insert(ar_nonce_limiter, {{vdf_server_peer, Peer}});
+				[{_, Peer}] ->
+					ok;
+				[{_, PreviousPeer}] ->
```

Comment:
Do you know when we'd expect this case to happen? Would it primarily be when a peer uses a domain name rather than IP address, and the IP underlying the domain name has changed?